hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/HoodieFileGroupReaderBasedParquetFileFormat.scala

```diff
@@ -152,8 +153,8 @@ class HoodieFileGroupReaderBasedParquetFileFormat(tablePath: String,
 
     // schema that we want fg reader to output to us
     val requestedSchema = StructType(requiredSchema.fields ++ partitionSchema.fields.filter(f => mandatoryFields.contains(f.name)))
-    val requestedAvroSchema = AvroConversionUtils.convertStructTypeToAvroSchema(requestedSchema, sanitizedTableName)
-    val dataAvroSchema = AvroConversionUtils.convertStructTypeToAvroSchema(dataSchema, sanitizedTableName)
+    val requestedAvroSchema = HoodieAvroUtils.repairSchema(AvroConversionUtils.convertStructTypeToAvroSchema(requestedSchema, sanitizedTableName), avroTableSchema)
```

Comment:
Instead of constructing a schema that needs repair, is it possible to just incorporate this into the initial conversion?

Reply:
+1, why not just fix the original conversion, is the Spark `StructType` missed something?

Reply:
yes, it only uses microseconds. Also it's missing enum type as well

Reply:
Hmm, I think `StructType` at least suppors timestamp with explicit precision? like timestamp(3) or timestamp(9)? Did you try that?